server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -83,6 +84,9 @@ public RemoteFsTranslog(
         try {
             download(translogTransferManager, location);
             Checkpoint checkpoint = readCheckpoint(location);
+            assert globalCheckpointSupplier instanceof ReplicationTracker
+                : "globalCheckpointSupplier is not instance of ReplicationTracker";
+            ((ReplicationTracker) globalCheckpointSupplier).updateGlobalCheckpoint(checkpoint.globalCheckpoint, "RemoteFsTranslog init");
```

Comment:
IMO, this update should happen in `IndexShard` instead of `RemoteFsTranslog`. As part of recovery, we can download the translog files and update the replicationTracker's global checkpoint.